src/bluesearch/database/download.py

```diff
@@ -103,15 +104,66 @@ def get_pmc_urls(
     return url_list
 
 
-def download_pmc_articles(url_list: list[str], output_dir: Path) -> None:
-    """Download PMC articles.
+def get_pubmed_urls(
+    start_date: datetime, end_date: datetime | None = None
+) -> list[str]:
+    """Get list of all PubMed incremental files to download.
+
+    Parameters
+    ----------
+    start_date
+        Starting date to download the incremental files (inclusive).
+    end_date
+        Ending date (inclusive). If None, today is considered as the ending date.
+
+    Returns
+    -------
+    list of str
+        List of all the PubMed urls (sorted chronologically).
+    """
+    if end_date is None:
+        end_date = datetime.today()
+
+    base_url = "https://ftp.ncbi.nlm.nih.gov/pubmed/updatefiles/"
+    r = requests.get(base_url)
+
+    entries = [line for line in r.text.split("\n") if '.gz"' in line]
+
+    pattern_date = re.compile("\\d{4}-\\d{2}-\\d{2}")
+    pattern_filename = re.compile("pubmed\\d{2}n\\d{4}.xml.gz")
+
+    matches_raw = []
+
+    for e in entries:
+        match_date = re.search(pattern_date, e)
+        match_filename = re.search(pattern_filename, e)
+
+        if match_date is not None and match_filename is not None:
+            matches_raw.append((match_date[0], match_filename[0]))
+
+    matches = [
+        (datetime.strptime(date_str, "%Y-%m-%d"), filename)
+        for date_str, filename in matches_raw
+    ]
+
+    matches_filtered = [m for m in matches if start_date <= m[0] <= end_date]
+    matches_sorted = sorted(matches_filtered, key=lambda m: m[0])
+
+    filenames = [base_url + filename for date, filename in matches_sorted]
+
+    return filenames
```

Comment:
I have a couple of suggestions:
* Use raw strigns to avoid double backslashes
* Match the file name and date in one match
* Use `match.groups()` to directly extract the matched parts

I tried with this snippet and it seemed to work. Not sure if there are any corner cases that need to be tested:
```python
    base_url = "https://ftp.ncbi.nlm.nih.gov/pubmed/updatefiles"
    response = requests.get(base_url)
    pattern = re.compile(r'<a href="(.*\.xml\.gz)">.*</a> *(\d{4}-\d{2}-\d{2})')
    urls = []
    for line in response.text.split("\n"):
        match = pattern.search(line)
        if not match:
            continue
        file_name, date_str = match.groups()
        date = datetime.strptime(date_str, "%Y-%m-%d")
        if start_date <= date <= end_date:
            urls.append(f"{base_url}/{file_name}")
```

Reply:
Very cool! I will try to put it in the source code:) The only thing I would like to point out is that your code does not have any sorting logic. However, maybe it does not really matter in what order we download the files. Also it seems like the response should be sorted chronologically already.

Reply:
Done in 6f80db1.

Reply:
Awesome. Just noticed though, you kept the previous `base_url`, which has a slash at the end. Then this line `urls.append(f"{base_url}/{file_name}")` will create a double slash in the middle. Not sure if this is a problem.

Reply:
Done in ecbe025